snapcraft/projects.py

```diff
@@ -713,6 +713,17 @@ def get_effective_base(self) -> str:
         if base is None:
             raise RuntimeError("cannot determine build base")
 
+        # "devel" is not a valid base but a valid build-base that may returned.
+        # the actual base is the same as the base defined in the project.
+        if base == "devel":
+            if allow_return_devel:
+                return base
+
+            if self.base is None:
+                raise RuntimeError("cannot determine base")
+
+            return self.base
+
```

Comment:
why not encode this logic in get_effective_base at https://github.com/snapcore/snapcraft/blob/d1fc926eb73d9483c64b75a9da99e49fe1530a03/snapcraft/utils.py#L210 ?